### PR TITLE
Added Attachment support to slash commands.

### PR DESCRIPTION
**Pull request by alentoghostflame**
## Summary

Adds the following to Slash:
* ApplicationCommandOptionType.attachment enum.
* Typehinting a parameter with `nextcord.Attachment` tells Discord that the parameter wants a file.
* Slash commands can now receive `Attachment` objects.

## Checklist

- [X] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [X] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: nextcord/enums.py [hunk 1]
```diff
@@ -575,6 +575,7 @@ class ApplicationCommandOptionType(Enum):
     role = 8
     mentionable = 9
     number = 10  # A double, AKA floating point.
+    attachment = 11
```
Comment (Skelmis) [inline nextcord/enums.py:578]: This may just be mobile. But is there a reason this is indented and the rest aren't?

**Review by Skelmis — APPROVED** [review]
Lgtm assuming tested

**Review by DenverCoder1 — APPROVED** [review]
Works great :heavy_check_mark:

**Review by TAG-Epic — APPROVED** [review]
Not tested

**Pull request closed without merging**